x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/cold_start_duration.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FAAS_COLDSTART_DURATION } from '../../../../../common/elasticsearch_fieldnames';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { fetchAndTransformMetrics } from '../../fetch_and_transform_metrics';
+import { ChartBase } from '../../types';
+
+const chartBase: ChartBase = {
+  title: i18n.translate('xpack.apm.agentMetrics.serverless.coldStartDuration', {
+    defaultMessage: 'Cold start duration',
+  }),
+  key: 'cold_start_duration',
+  type: 'linemark',
+  yUnit: 'time',
+  series: {
+    coldStart: {
+      title: i18n.translate(
+        'xpack.apm.agentMetrics.serverless.coldStartDuration',
+        { defaultMessage: 'Cold start duration' }
+      ),
+    },
+  },
+};
+
+export function getColdStartDuration({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  serviceNodeName,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  serviceNodeName?: string;
+  start: number;
+  end: number;
+}) {
+  return fetchAndTransformMetrics({
```

Comment:
I'm not sure if we should keep this abstraction. If you find it useful we can keep it but if not I'd rather have more duplication (or a better abstraction).

Reply:
I think it's fine, it does what it should do and I think it was very easy to add new APIs using it.

Reply:
FWIW, the `fetchAndTransformMetrics` function expects a single value. That means we have to use a scripted aggregation as you can see below in `computeUsageScript`. Did you compare performance of running several aggs and calculating the result kibana-side with the scripted avg aggregation? Scripted aggs are slower and not cachable.

Reply:
No, I did not. But I see your point. I'll fetch the data and calculate it in kibana and see how it works.

Reply:
Fixed the compute usage here https://github.com/elastic/kibana/pull/139041/commits/b14a5f5b1e6409c1b94318196dcb888a6623054c, removed the script and calculate the value in the server instead